### PR TITLE
fix: fields weight string to integer

### PR DESCRIPTION
**Pull request by maxwelbm**
This pull request addresses the API update that involved changing the type of the weight field from string to integer. Previously, the weight field was set to string, which caused errors when dealing with integer values. The SDK was updated to reflect this change and then the field was set to integer in the corresponding plugin.

### CODE REVIEW

**Review by pablodiehl — DISMISSED** [review]

File: main.go [hunk 1]
```diff
@@ -25,6 +26,8 @@ var (
 )
 
 func main() {
+	log.Println("Garantindo que isto está rodando agora a partil de agora: ", time.Now().Hour(), " - ", time.Now().Minute())
```
Comment (PatrickMenoti) [inline main.go:29]: oops

File: internal/data_source_edge_applications_origins.go [hunk 2]
```diff
@@ -268,13 +269,15 @@ func (o *OriginsDataSource) Read(ctx context.Context, req datasource.ReadRequest
 		next = *originsResponse.Links.Next.Get()
 	}
 
+	log.Println("HERE 01")
```
Comment (PatrickMenoti) [inline internal/data_source_edge_applications_origins.go:272]: oops

File: func-tests/main.tf [hunk 3]
```diff
@@ -183,27 +183,27 @@ resource "azion_edge_function" "testfunc2firewall" {
...
-
-  provisioner "local-exec" {
-    command = "sleep 20 && terraform apply -auto-approve -target='azion_edge_application_main_setting.testfunc' -var 'edge_functions_module=true'"
-  }
-}
-
-resource "azion_edge_application_edge_functions_instance" "testfunc" {
-  depends_on = [null_resource.update_edge_functions]
+# resource "null_resource" "update_edge_functions" {
+#   depends_on = [azion_edge_application_main_setting.testfunc]
```
Comment (PatrickMenoti) [inline func-tests/main.tf:187]: don't we need these commented code sections?

Comment (maxwelbm) [reply func-tests/main.tf:187]: Currently I want to leave these comments to understand if we can uncomment this code and use it as an example for new tests

**Review by rodrigomaria — APPROVED** [review]

**Review by PatrickMenoti — APPROVED** [review]

**Pull request merged**